handler/src/main/java/io/netty/handler/ssl/Java9SslUtils.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import io.netty.util.internal.PlatformDependent;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLParameters;
+import java.lang.reflect.Method;
+import java.util.List;
+
+final class Java9SslUtils {
+    private static final Method setApplicationProtocols;
+    private static final Method getApplicationProtocol;
+
+    static {
+        Method setter = null;
+        Method getter = null;
+
+        if (PlatformDependent.javaVersion() >= 9) {
+            try {
+                getter = SSLEngine.class.getMethod("getApplicationProtocol");
+                setter = SSLParameters.class.getMethod("setApplicationProtocols", String[].class);
+            } catch (Throwable t) {
+                throw new RuntimeException("Unable to initialize Java9SslUtils", t);
```

Comment:
I think this may be a bit to aggressive as these methods could also throw if a SecurityManager is in place or I am wrong ?